### PR TITLE
feat: relative/absolute effectivity thresholds for QLaw

### PR DESCRIPTION
**Pull request by vishwa2710**
This allows us to trade time of flight with fuel mass. It will force some coast arcs so that we burn at "optimal" points in the orbit.

Screen Shot 2023-12-01 at 9 22 38 AM

### CODE REVIEW

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 1]
```diff
@@ -2738,6 +2738,8 @@ TEST_P(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator_QLaw, Q
 
     const COE endCOE = COE::Cartesian({state.getPosition(), state.getVelocity()}, gravitationalParameter);
 
+    std::cout << endCOE << std::endl;
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2741]: TBR?

File: src/OpenSpaceToolkit/Astrodynamics/GuidanceLaw/QLaw.cpp [hunk 2]
```diff
@@ -636,6 +676,51 @@ Vector5d QLaw::computeDeltaCOE(const Vector5d& aCOEVector) const
     };
 }
 
+Tuple<double, double> QLaw::computeEffectivity(
+    const Vector6d& aCOEVector, const Vector3d& currentThrustDirection, const Vector5d& dQ_dOE
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/GuidanceLaw/QLaw.cpp:680]: Maybe just add comment with paper and section? 🙏

**Review by Derollez — APPROVED** [review]
Stellar

File: src/OpenSpaceToolkit/Astrodynamics/GuidanceLaw/QLaw.cpp (hunk 2 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/GuidanceLaw/QLaw.cpp:680]: 💯 this is a great idea. I'll add some details as this was not trivial for me to find as well.

**Pull request merged**